addons/web/static/tests/fields/relational_fields_tests.js

```diff
@@ -2165,8 +2165,9 @@ QUnit.module('relational_fields', {
 
         await testUtils.form.clickSave(form);
 
-        assert.strictEqual(form.$('.o_field_widget').text(), 'Black',
-            "value should be 'Black'");
+        assert.strictEqual(form.$('.o_field_widget').text().trim().split(/\s+/g).join(','), 'Red,Black');
+        assert.containsOnce(form, '.o_radio_input[data-index=1]:checked',
+            "'Black' should be checked");
```

Comment:
the diff in tests looks a bit weak with respect to the diff in the code. Some changes are probably worth new tests (like the many2one multi line stuff, and the Add a line in x2many)